[WooExpress Removal] Remove store creation from the login flow

hichamboushaba opened the pull request:
<!-- Remember about a good descriptive title. -->

Part of: #11115 
<!-- Id number of the GitHub issue this PR addresses. -->

⚠️ please don't merge yet, I'll merge everything when all PRs are approved.

### Description
This is the first part of WooExpress sign-up removal, it touches the login part, and has the following changes:
1. Update the prologue screen to remove the store creation option and replace it with a tertiary button to open a webpage.
2. Remove the "Create account" button from the wrong email screen.
3. Remove the store creation option from the site picker.

**This PR doesn't remove any logic code, the code cleanup will be handled separately.**

### Testing instructions
1. Check the login prologue, and confirm that it has a text button "Starting a new store?"
2. Click on the button, and confirm it opens the following [webpage](https://woo.com/hosting-solutions/)
3. Go back, and click on Login.
4. Enter the address of a Jetpack site.
5. In the email screen, enter a non valid email address.
6. In the "Email not found" screen, confirm that the "Create an account" button was removed.
7. Go back, and continue signing with a valid WordPress.com account.
8. Open the site picker.
9. Click on "Add a store"
10. Confirm it opens directly a screen for connecting an existing store. 

### Images/gif
<img width=360 src="https://github.com/woocommerce/woocommerce-android/assets/1657201/7f915b97-c41d-4511-96c2-3df1835d779a" />


- [ ] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

hichamboushaba committed:
Add missing import

hichamboushaba committed:
Remove unused code

hichamboushaba committed:
Remove store creation button from prologue and replace it with a webpage

hichamboushaba committed:
Remove "create an account" button from the wrong email screen

hichamboushaba committed:
Remove store creation option from the site picker

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  1 Message: <p>This PR contains changes to...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/202">17.9</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false"><p>This PR contains changes to Tracks-related logic. Please ensure (<strong>author and reviewer</strong>) the following are completed:</p>

<ul>
  <li>The tracks events must be validated in the Tracks system.</li>
  <li>Verify the internal Tracks spreadsheet has also been updated.</li>
  <li>Please consider registering any new events.</li>
  <li>The PR must be assigned the <strong>category: tracks</strong> label.</li>
</ul></td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

hichamboushaba committed:
Remove listener to fix detekt issue

JorgeMucientes commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/LoginPrologueFragment.kt:
```diff
@@ -52,17 +49,14 @@ open class LoginPrologueFragment(@LayoutRes layout: Int) : Fragment(layout) {
                 AnalyticsTracker.track(
-                    AnalyticsEvent.LOGIN_PROLOGUE_CREATE_SITE_TAPPED,
-                    mapOf(AnalyticsTracker.KEY_IS_FREE_TRIAL to true)
+                    AnalyticsEvent.LOGIN_PROLOGUE_CREATE_SITE_TAPPED
```
Hmm, I'm thinking we might want to use a different event for this button to easily differentiate between the "Create Store" from the "Starting a new store" ?

hichamboushaba replied:
Good point, updated in 579c8e8ef68e8cbdc8e2e48535a209e95602451a

JorgeMucientes commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/login/error/LoginNoWPcomAccountFoundDialogFragment.kt:
```diff
@@ -41,17 +37,6 @@ class LoginNoWPcomAccountFoundDialogFragment : LoginBaseErrorDialogFragment() {
-        get() = LoginErrorButton(
-            title = R.string.login_create_an_account,
-            onClick = {
-                appPrefsWrapper.setStoreCreationSource(AnalyticsTracker.VALUE_LOGIN_EMAIL_ERROR)
```
I think this tracking value: `VALUE_LOGIN_EMAIL_ERROR` is unused now and can be removed.

hichamboushaba replied:
Code cleanup will come in a separate PR.

JorgeMucientes commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductListFragment.kt:
```diff
@@ -29,6 +29,7 @@ import com.woocommerce.android.databinding.DialogProductListBulkPriceUpdateBindi
 import com.woocommerce.android.databinding.FragmentProductListBinding
 import com.woocommerce.android.extensions.handleDialogResult
 import com.woocommerce.android.extensions.handleResult
+import com.woocommerce.android.extensions.navigateSafely
```
???? Where's this coming from and how come it wasn't there already? 😅

hichamboushaba replied:
I think there were two PRs merged around the same time where one removed all usages of `navigateSafely` and one added a new one, which caused the import to be accidentally removed and breaking trunk builds.

JorgeMucientes commented:
Nicely done @hichamboushaba everything works as expected and code looks good. 

I have a couple of thoughts about the new UX I'd like to share: 

- `Add a store` text in site picker. This text would give me the impression I can create a "new" store I can connect to. Wdyt about using a different copy? Something like `Connect another store`. Something that makes it clear that the purpose of that CTA is to connect to an existing store, not creating a new one. I know in the next screen this is clear already, but making the CTA's purpose clearer wouldn't harm. 

- In the LoginPrologue, we are showing a text about a 15-day free trial that feels misleading now that we are removing WooExpress. The users will never find this "Free trial" thing in the app and is unlikely they'll find it in WooCommerce experience at all. I'd suggest discussing this with @joe-keenan and come up with some different copy. 

![Screenshot 2024-03-21 at 10 09 53](https://github.com/woocommerce/woocommerce-android/assets/2663464/f314024e-e870-48aa-a238-e185bfee75d9)

Let me know your thoughts :)

hichamboushaba committed:
Reword the Add a store button in the Site picker

hichamboushaba committed:
Restore prologue's text before adding store creation

The existing one now mentions store creation, and the app doesn't support it anymore.

hichamboushaba committed:
Rename Tracks event

hichamboushaba commented:
Thank you @JorgeMucientes for the review, both good points, I missed those. I did the following changes:

1. I updated the site's picker button to `Connect another store`.
2. I updated the prologue's text to `From your first sale to millions in revenue, Woo is with you. See why merchants trust us to power 3.4 million online stores`, which is what we were using before. I'll still ask Joe about his opinion and update if needed.

JorgeMucientes commented:
Thanks for applying the changes @hichamboushaba, ready to 🚢 now!
